basis/cli/commands/create.py

```diff
@@ -95,8 +91,49 @@ def node(
             name=name or location.stem, node_file=node_file, id=str(NodeId.random()),
         )
 
-    # Write to files last to avoid partial updates
-    location.write_text(content)
+    # Write to disk last to avoid partial updates
+    if component:
+        with abort_on_error("Adding component failed"):
+            search = (r for r in graph_components_all() if r["name"] == component)
+            component_resp = next(search, None)
+            if not component_resp:
+                sprint(f"[error]No component named {component}")
+                sprint(
+                    f"[info]Run [b]basis list components[/b] "
+                    f"for a list of available components"
+                )
+                raise typer.Exit(1)
+
+            zip_src_path = component_resp["file_path"]
+            if zip_src_path.endswith("graph.yml") and location.name != "graph.yml":
+                sprint(f"[error]Component {component} is a subgraph")
+                norm = re.sub(r"\W", "_", component)
+                suggestion = f"{norm}/graph.yml"
+                sprint(
+                    f"\n[info]For subgraph components, you need to pass the "
+                    f"location of the graph.yml file to create"
+                )
+                sprint(
+                    f"[info]Try [code]basis create node "
+                    f"--component='{component}' {suggestion}"
+                )
+                raise typer.Exit(1)
+
+            b = io.BytesIO(download_graph_zip(component_resp["graph_version_uid"]))
+            dir_editor = GraphDirectoryEditor(ids.graph_file_path, overwrite=False)
+            with ZipFile(b, "r") as zf:
+                dir_editor.add_node_from_zip(zip_src_path, location, zf)
+    else:
+        if location.suffix == ".py":
+            fun_name = re.sub(r"[^a-zA-Z0-9_]", "_", location.stem)
+            if re.match(r"\d", fun_name):
+                fun_name = f"node_{fun_name}"
+            content = _PY_FILE_TEMPLATE.format(fun_name)
+        elif location.suffix == ".sql":
+            content = _SQL_FILE_TEMPLATE
+        else:
+            abort("Node file location must end in .py or .sql")
```

Comment:
Could be a chart.json too?

Reply:
not right now; we don't have a template for charts. We can add that in a separate pr. (this code path is only for creating new nodes, not for components)